staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -541,9 +537,48 @@ func (g *Cloud) ensureInternalInstanceGroup(name, zone string, nodes []*v1.Node)
 func (g *Cloud) ensureInternalInstanceGroups(name string, nodes []*v1.Node) ([]string, error) {
 	zonedNodes := splitNodesByZone(nodes)
 	klog.V(2).Infof("ensureInternalInstanceGroups(%v): %d nodes over %d zones in region %v", name, len(nodes), len(zonedNodes), g.region)
+
 	var igLinks []string
-	for zone, nodes := range zonedNodes {
-		igLink, err := g.ensureInternalInstanceGroup(name, zone, nodes)
+	gceZonedNodes := map[string][]string{}
+	for zone, zNodes := range zonedNodes {
+		hosts, err := g.getFoundInstanceByNames(nodeNames(zNodes))
+		if err != nil {
+			return nil, err
+		}
+		names := sets.NewString()
+		for _, h := range hosts {
```

Comment:
Can we just loop through zNodes and use node.Name? Wouldn't it be the same?

Reply:
So if there is an GCE Instance named `worker-0`, the internal dns name for that instance is `worker-0.<zone>.c.<project name>`
some linux OSes will use that to set the hostname to `worker-0` but some of them will keep the entire internal DNS name.
But the kubelet uses hostname to set the name of the node.

so you might receive a list of `worker-{...}.<zone>.c.<project name>`, but you want to deal with instance names ie `worker={..}` when interacting with GCE apis, therefore i'm iterating over hosts/instances.Name instead of nodes.Name